demos/components/src/components/Notification/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+import { StyledNotification, StyledButton } from './Styled';
+import { Caution, Check, Information } from '../icons';
+import { Button } from '../Button';
+
+export type Severity = 'success' | 'warning' | 'error' | 'info';
+export type Size = 'sm' | 'md' | 'lg' | 'fill';
+
+export interface NotificationProps extends React.HTMLAttributes<HTMLDivElement> {
+  icon?: React.ReactNode;
+  severity?: Severity;
+  onClose?: () => void;
+  closeText?: string;
+  size?: Size;
+}
+
+const iconMapping = {
+  success: <Check />,
+  warning: <Caution />,
+  error: <Caution />,
+  info: <Information />,
+};
+
+export const Notification: React.FC<NotificationProps> = props => {
+  const {
+    closeText = 'close',
+    severity = 'info',
+    size = 'md',
+    icon = iconMapping[severity],
+    onClose,
+    children,
+  } = props;
+  const ariaLive = severity === 'error' ? 'assertive' : 'polite';
+  const ariaRole = severity === 'error' ? 'alert' : 'status';
+
+  return (
+    <StyledNotification
+      aria-live={ariaLive}
+      role={ariaRole}
+      severity={severity}
+      size={size}
+      {...props}
+    >
+      <div className="icon">{icon}</div>
+      <div className="message">{children}</div>
+      {onClose && (
+        <Button
+          className="button"
+          addStyles={StyledButton}
+          onClick={() => onClose()}
```

Comment:
Passing a new function inline will keep force `<Button>` to re-render even if a parent component passes a memoized callback. Either pass `onClose` directly or created a memorized callback.

Reply:
Good point, will update. Originally I felt that re-rendering would be fairly inexpensive here, but for best practice I should have done it that way anyways